src/server/kiwix-serve.cpp

```diff
@@ -324,18 +324,12 @@ int main(int argc, char** argv)
   auto libraryFileTimestamp = newestFileTimestamp(libraryPaths);
   auto curLibraryFileTimestamp = libraryFileTimestamp;
 
-  /* Infer ipMode from address */
-  kiwix::IpMode ipMode = kiwix::IpMode::ipv4;
-
-  if (address == "all"){
-    address = "";
-    ipMode = kiwix::IpMode::all;
-  } else if (address == "ipv4"){
-    address = "";
-  } else if (address == "ipv6"){
-    address = "";
-    ipMode = kiwix::IpMode::ipv6;
-  }
+  kiwix::IpMode ipMode =  (address == "all")   ? kiwix::IpMode::all  :
+                          (address == "ipv6")  ? kiwix::IpMode::ipv6 :
+                                                 kiwix::IpMode::ipv4;
```

Comment:
Essentially the logic remains the same as before. How it is setup is that the `InternalServer::start` is going to set the `m_ipMode` as ipv6 (previously all) if a valid v6 ip has been provided by the user.

Reply:
Maybe. But codewise it appears flawed. I think that the interaction of `Server::setIpMode()` and `Server::setAddress()` should be defined in such a way that the former takes precedence. In other words, if an IPv4 mode is requested then only IPv4 addresses should be accepted.

Reply:
So essentially we should:
- Remove all server mode handling and move it to kiwix-serve (removing also [kiwix-serve.cpp:381](https://github.com/kiwix/kiwix-tools/pull/704/files#diff-29eedb326cc54203c610d3910176face715ff30664857e7d8ced287ff455551aR381)).
- Set default mode (no mode through `-i` provided) to `all` (also, [server.h:84](https://github.com/kiwix/libkiwix/blob/0b14fda94d6d7047bbcc9b00cb53f261fcd66880/include/server.h#L84) should be changed).

Correct?

Reply:
> * Remove all server mode handling and move it to kiwix-serve

I don't know what is covered by this so cannot answer

> * Set default mode (no mode through `-i` provided) to `all`

It will work, but IMHO it's better to add an extra `enum` member `IpMode::auto`

Reply:
Would `IpMode::auto` have any use case other than `IpMode::all`'s?

Reply:
Or just an easier way for kiwix-serve to understand that no ip mode has been supplied?

Reply:
> Would `IpMode::auto` have any use case other than `IpMode::all`'s?

It was included in my [original proposal](https://github.com/kiwix/libkiwix/pull/1074#pullrequestreview-2019253469) for enhancing the `kiwix-serve` CLI with support for IPv6, but did not make way into implementation. Whatever the reason for that (you can find it in the ensuing discussion) it has a comeback at least in the C++ code.

> Or just an easier way for kiwix-serve to understand that no ip mode has been supplied?

Mostly yes, but with some potential to find other uses later.

Reply:
Alright, I see how it can be useful in future scenarios. I will add it and make it the default mode for when an address is provided or no -i at all.

P.S. `auto` is not allowed as a name. Will use `flex`.

Reply:
To be honest it is a bit complicated with the current implementation to really put it in use, but I will wait for your next review after committing to discuss further.